### DISCUSSION TITLE
SessionToken lifetime AssumeRoleWithLDAPIdentity method

### DISCUSSION THREAD

**Discussion by pternyuk** [minio/minio#17219] [Q&A] [answered]
Hi team,
we have business need to interact with minIO via spark with LDAP authentication.
I use AssumeRoleWithLDAPIdentity method to get temporary credentials and then use them to create spark session. But it is not suitable for long running spark jobs, cause sessionToken is valid only for 1 hour.
Other ways ot interact with STS API have DurationSeconds param to set sessionToken lifetime, but AssumeRoleWithLDAPIdentity method doesn't have it.
Is there any way to increase sessionTocken lifetime for AssumeRoleWithLDAPIdentity method?
Or maybe there is another way to interact with minIO via spark with LDAP authentication in long running jobs?

Thanks in advance.

**Comment by harshavardhana** [comment, marked as answer]
The default lifetime of LDAP identity is governed via DurationSeconds where do you see that DurationSeconds doesn't exist?

**Reply by pternyuk** [reply]
According to documentation we have 3 way to get temporary credentials (https://min.io/docs/minio/linux/developers/security-token-service.html)
AssumeRoleWithWebIdentity and AssumeRoleWithCustomToken have DurationSeconds param to set sessionToken lifetime, but AssumeRoleWithLDAPIdentity method doesn't have it (https://min.io/docs/minio/linux/developers/security-token-service/AssumeRoleWithLDAPIdentity.html#minio-sts-assumerolewithldapidentity).
But now i tried make sample code to check and method AssumeRoleWithLDAPIdentity accepted param DurationSeconds
Thank you for your answer!

def get_credentials(endpoint, username, password):
    params = {
    "Action": "AssumeRoleWithLDAPIdentity",
    "LDAPUsername": username,
    "LDAPPassword": password,
    "Version": "2011-06-15",
    "DurationSeconds": "7200"
        }
    r = requests.post(endpoint, params=params)
    credentials = {}
    content = r.content
    root = etree.fromstring(content)
    ns = {"ns": "https://sts.amazonaws.com/doc/2011-06-15/"}
    et = root.find("ns:AssumeRoleWithLDAPIdentityResult/ns:Credentials", ns)
    for el in et:
        _, _, tag = el.tag.rpartition("}")
        credentials[tag] = el.text
    return credentials

**Reply by ravindk89** [reply]
I think this is just an accident in our docs - likely the field was dropped as we were doing some migration work.

https://github.com/minio/minio/blob/master/docs/sts/ldap.md#durationseconds

We will update the docs appropriately.
